scripts/sync_package_version.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Update the version of a given package in the related package.json file
+# under "./examples" and "../mephisto/abstractions/blueprints"
+# Currently we support the update of the following packages:
+# * mephisto-task
+# * bootstrap-chat
+
+# Usage:
+# You can sync one package at a time like this:
+# ./sync_package_version.sh mephisto-task 1.0.13
+# ./sync_package_version.sh bootstrap-chat 1.0.7
+
+# You can also sync all packages together like this:
+# ./sync_package_version.sh mephisto-task 1.0.13 bootstrap-chat 1.0.7
+
+possible_dirs=`find ../mephisto/abstractions/blueprints ../examples -type d \( -name node_modules -o -name tmp -o -name runs -o -name _generated \) -prune -false -o -name 'package.json' -exec dirname {} \;`
+
+
+sync_package_version() {
+    package_name=$1
+    new_version=$2
+
+    for dir in $possible_dirs
+    do
+        (cd "$dir" && pwd && sed -i '' "s/\"$package_name\":.*/\"$package_name\": \"$new_version\",/" package.json)
```

Comment:
**About the pwd:**
Nice catch :) It is a little trick to print out the directories that has been touched. As a user, I feel safer when I see what I has been changed. For example:
py```
/Users/liusha/repos/Mephisto/mephisto/abstractions/blueprints/parlai_chat/webapp
/Users/liusha/repos/Mephisto/mephisto/abstractions/blueprints/static_html_task/source
/Users/liusha/repos/Mephisto/examples/parlai_chat_task_demo/webapp
/Users/liusha/repos/Mephisto/examples/static_react_task/webapp
All mephisto-task in the files above are updated to version 1.0.13
```

**About autoupdating the packages:**
It seems that this feature is resolving a different, updating the packages, which is out of the scope of this issue. I think it is the best to create another task for that feature if we need it :) 

If we want to implement this feature at least there are a few things we need to define:
* what if there is no internet connection
* what if the update failed and how to fallback
* should this feature be implemented in this script specifically? The logic of updating an out-of-date package is likely be used by another script/feature. Will it be a better way if we implement that feature in a different so that the logic can be reused? 
* ...

Let me know what you think :)

Reply:
I could see two use-cases for this script:
- One is for library maintainers like ourselves to update subdependencies when we push out an updated package.
- The other is for library users to update their forked repo to use the latest packages.

I think doing `npm install $package_name@$new_version` only will help the second use-case, since we do not commit the `node_modules/` folder in our repo. The one exception that would make sense for the first use-case, is that it would update `yarn.lock` and `package-lock.json` files though, though I'm not sure if we really use that at the moment.

If we're prioritizing the first use-case, maybe just a find-replace is fine given that `node_modules` isn't checked in?

--

Can't speak for @liushh but maybe the `pwd` is just for UI progress/feedback and printing out which folder it is currently working on?